### ISSUE TITLE
Fix repacking

### ISSUE THREAD

**Issue by rfuzzo** [rfuzzo/CP77Tools#53]
Repacking currently only works on (some) files with no buffers. What needs to be done:

- [x] **Figure out how the SHA1 hash in Table1 of an archive is calculated for buffered files.** For files with no buffers, it is SHA1 over the uncompressed file. For files with buffers, it is currently unknown. Code link: https://github.com/WolvenKit/CP77Tools/blob/ea64254f9329ec7250ca9a255973bb253804344c/CP77.CR2W/Archive/Archive.cs#L237

- [x] Files with buffers consist of a cr2w main file (e.g. .xbm) and one or more .buffer files. The main cr2w file has a table (table 3, or Buffer table) with references to the buffer files inside, complete with size and zsize of the buffer as well as offset. When packing (or editing the cr2w file) **the buffer table has to be adjusted,** as well as the **crc of the cr2w file in its header.** https://github.com/WolvenKit/CP77Tools/blob/ea64254f9329ec7250ca9a255973bb253804344c/CP77.CR2W/CR2W/CR2WBuffer.cs#L10

**Comment by zao** [comment]
Recording it here as well for permanence - for the SHA1 hashes for buffered files I've found that the hash is obtained by hashing first the contents of the uncompressed main file, followed by the possibly compressed bytes of the buffers in buffer order.

Using the JSON (`archive -d`), extracted files (`archive -e`) and actual `.archive` files from my GOG 1.06 release I've verified this for all entries in the JSON files but one. The only exception found was the hidden message video which lacks buffers and just seems to have an incorrect hash recorded.
